modules/Emails/EmailsController.php

```diff
@@ -171,9 +171,13 @@ public function action_send()
     {
         global $current_user;
         global $app_strings;
+        global $sugar_config;
 
         $request = $_REQUEST;
 
+        // Set message_id for email record in DB
+        $request['message_id'] = 'SUITECRM_' . md5('HELLO'.(idate("U")-1000000000).uniqid()) . '@' . $sugar_config['host_name'];
```

Comment:
Calling uniqid() without any parameter is a bit dangerous ans not a best option as it's related on a system time clock.
see more in php.net: 
> Warning
This function does not guarantee uniqueness of return value. Since most systems adjust system clock by NTP or like, system time is changed constantly. Therefore, it is possible that this function does not return unique ID for the process/thread. Use more_entropy to increase likelihood of uniqueness.
http://php.net/manual/en/function.uniqid.php#120123

Reply:
I'm trying to implement open and click tracking with this upcoming feature.

My use case is that I'm using Mailgun that provides callbacks when these emails are opened. Mailgun refers back to emails which were acted upon by their `Message-Id`. If the hostname isn't part of that value which is saved to the DB, an additional processing step is needed to split that value by `@` and do the lookup based on the first part.

From this perspective, it would be more straightforward, if the value of the `message_id` field in the DB was exactly the same as the value of the `Message-Id` field in sent emails.

Currently I'm using this in my custom SuiteCRM installation:

```
$request['message_id'] = '<' . time().'.'.bin2hex(random_bytes(8)).'@'.gethostname().'>';
```

This produces a value, something like `<1549445373.02231e5c88e1a743@typhoon>`. 

What do you think @ApatheticCosmos @gymad?